src/amltk/optimization/trial.py

```diff
@@ -130,13 +130,53 @@ def target_function(trial: Trial) -> Trial.Report:
             options:
                 members: False
 
-    Some important properties is that they have a unique
+    Some important properties are that they have a unique
     [`.name`][amltk.optimization.Trial.name] given the optimization run,
-    a candidate [`.config`][amltk.optimization.Trial.config]' to evaluate,
+    a candidate [`.config`][amltk.optimization.Trial.config] to evaluate,
     a possible [`.seed`][amltk.optimization.Trial.seed] to use,
-    and an [`.info`][amltk.optimization.Trial.info] object which is the optimizer
+    and an [`.info`][amltk.optimization.Trial.info] object, which is the optimizer
     specific information, if required by you.
 
+    !!! tip
+    When using the `success` method, make sure to provide values for all metrics
+    specified in the `metrics` attribute. Each metric has a unique name, and it's
+    crucial to use the correct names when reporting success. Failure to provide
+    a value for any metric will result in a `ValueError`. To ensure correctness,
+    refer to the `self.metrics` attribute for the list of metrics and use their
+    exact names in the `trial.success` method. For example:
+
+    ```python
+    trial = Trial(
+        name="example_trial",
+        config={"param": 42},
+        metrics=[Metric(name="accuracy", minimize=False)]
+    )
+
+    # Incorrect usage (will raise a ValueError)
+    report = trial.success(invalid_metric=0.95)
+
+    # Correct usage
+    report = trial.success(accuracy=0.95)
+    ```
+
+    Additionally, if you encounter a metric name mismatch,
+    consider renaming the metric using the provided key:
+
+    ```python
+    trial = Trial(
+        name="example_trial",
+        config={"param": 42},
+        metrics=[Metric(name="poly_metric", minimize=False)]
+    )
+
+    # Incorrect usage (will raise a ValueError)
+    report = trial.success(poly_metric=0.95)
+
+    # Correct usage
+    report = trial.success(provided_key=0.95)
+    trial.metrics = [Metric(name="provided_key", minimize=False, bounds=None)]
```

Comment:
You should neveer really change the metrics for the trial, they're set by the Optimizer. I'll update the docs to reflect this